shaders/commonProfileShaders.cpp

```diff
@@ -55,11 +56,17 @@ namespace GLTF
         ((std::istream*)png_get_io_ptr(pngPtr))->read((char*)data, length);
     }
 #endif
+	static std::unordered_map<std::string, bool> haveAlphas;		// Caching to remove up to 98% execution time
```

Comment:
Usually, I'm hesitant to bring up this one, certain quote by Donald Knuth, but considering the fact that there will rarely be *ten million textures* in one model, it might be appropriate in this case. Reading a single byte via network or HDD will take longer than thousands of map lookups.

**But**, admittedly, there *may* be an issue: 

Visual Studio has some performance problems with `unordered_` data structures in debug mode. And here, I'm not talking about milliseconds, but about timings that are ridiculous, or "pathological", as they are called on http://www.drdobbs.com/cpp/c11-hash-containers-and-debug-mode/232200410?pgno=2 (This is just one reference - I also experienced this in practice. Calling the destructor of an unordered map with a few thousand elements can indeed take *several seconds*....). 

Workarounds may involve a `DEBUG`-flag dependent `typedef` for the map type, or simply resorting to `map`...